Assets/AWSIM/Scenes/Main/AutowareSimulation/AutowareSimulation.cs

```diff
@@ -62,6 +65,19 @@ void Awake()
 
                 var rotation = Quaternion.Euler(config.Ego.EulerAngles);
                 egoTransform.rotation = ROS2Utility.RosToUnityRotation(rotation);
+
+                // set time source
+                timeSourceSelector?.SetType(config.TimeSource);
+            }
+        }
+
+        void CollectComponents()
+        {
+            // get time source selector
+            timeSourceSelector = FindObjectOfType<TimeSourceSelector>();
```

Comment:
Instead of using FindObjectOfType, could you please add a [SerializeField] TimeSourceSelector to AutowareSimulation.cs?
I would like to avoid type find at runtime as much as possible.

Reply:
Good idea. I've just made the change. Thanks